### PR TITLE
Use proper HostConfig element which contains the proper Memory value …

### LINKED ISSUE #1766: Docker module gives error KeyError: 'Memory' when using  remote API v1.19
**Summary**

When running a playbook with _docker_api_version: "1.19"_, the function _get_differing_containers_ gives the following error:

```
Traceback (most recent call last):
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437502622.78-176169208388238/docker", line 3132, in <modu
le>
    main()
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437502622.78-176169208388238/docker", line 1496, in main
    reloaded(manager, containers, count, name)
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437502622.78-176169208388238/docker", line 1373, in reloa
ded
    for container in manager.get_differing_containers():
  File "/home/micheal/.ansible/tmp/ansible-tmp-1437502622.78-176169208388238/docker", line 908, in get_di
ffering_containers
    actual_mem = container['Config']['Memory']
KeyError: 'Memory'
```

If  _docker_api_version: "1.18"_ is used it does not give an error and deploys the container successfully. It appears it could be related to a change in Remote API v1.19 where,

> The fields Debug, IPv4Forwarding, MemoryLimit, and SwapLimit are now returned as boolean instead of as an int.
> -[Docker Remote API v1.19 Changes](https://docs.docker.com/reference/api/docker_remote_api/#v1-19)

**Ansible Version**

```
 ansible 1.9.2
```

**Ansible Configuration**

```
 host_key_checking = False
 scp_if_ssh = True
```

**Playbook**

```
  - name: Deploy NGINX container
    docker:
      name: nginx
      registry: "private-docker-registry"
      docker_api_version: "1.19"
      pull: always
      tty: yes
      image: private-docker-registry/nginx
      state: reloaded
      net: bridge
      ports:
      - "0.0.0.0:8090:8090"
      volumes:
      - /mnt/nginx/:nginx:ro
```

**Environment**
- Ubuntu 14.04 ansible master
  - ansible v1.9.2
- CentOS 6.6 container host 
  - Docker v1.7.1 
  - docker-py v1.3.0

**Steps to Reproduce**
- Install Ansible v1.9.2 on Ubuntu 14.04
- Install Docker v1.7.1 and docker-py v1.3.0 on CentOS 6.6
- Create a playbook with _docker_api_version: "1.19"_ to use Remote API version 1.19
- Run playbook on Ubuntu ansible host to deploy a container on the CentOS container host

### PR DESCRIPTION
**Pull request by ecliptik**
…- fixes #1766 

Tested locally and containers deployed properly using _docker_api_version: "1.19"_

### CODE REVIEW

**Comment by bcoca** [comment]
what happens to those using earlier versions? this should check api version and use one or the other

**Comment by bcoca** [comment]
as these show, this is now a conflict between those running diff versions #1744 #1763 #1602 #1766

**Comment by ecliptik** [comment]
Updated the pull request to include an API check and use the proper data structure depending on the API version,
- v1.18 and earlier - ['Config']['Memory']
- v1.19 -  ['HostConfig']['Memory']

This still gives the error which is due to #1707 and looks like it should work in Ansible v1.9.3 or v2.0.

```
  File "/usr/lib/python2.6/site-packages/docker/utils/utils.py", line 546, in create_container_config
    'mem_limit has been moved to host_config in API version 1.19'
docker.errors.InvalidVersion: mem_limit has been moved to host_config in API version 1.19
```

File: cloud/docker/docker.py [hunk 1]
```diff
@@ -1008,7 +1011,11 @@ def get_differing_containers(self):
             except ValueError as e:
                 self.module.fail_json(msg=str(e))
 
-            actual_mem = container['Config']['Memory']
+            #For v1.19 API and above use HostConfig, otherwise use Config
+            if docker_api_version >= 1.19:
+                actual_mem = container['HostConfig']['Memory']
+            else:
+                actual_mem = container['Config']['Memory']
```
Comment (bcoca) [inline cloud/docker/docker.py:1018]: is it not possible to use ensure_capability as we do with other features?

**Pull request merged**